tests/pkg/notifications/asserters/formation_assignments_async.go

```diff
@@ -14,52 +15,85 @@ import (
 
 type FormationAssignmentsAsyncAsserter struct {
 	FormationAssignmentsAsserter
-	delay int64
+	timeout time.Duration
+	tick time.Duration
 }
 
-func NewFormationAssignmentAsyncAsserter(expectations map[string]map[string]fixtures.AssignmentState, expectedAssignmentsCount int, certSecuredGraphQLClient *graphql.Client, tenantID string, delay int64) *FormationAssignmentsAsyncAsserter {
+func NewFormationAssignmentAsyncAsserter(expectations map[string]map[string]fixtures.AssignmentState, expectedAssignmentsCount int, certSecuredGraphQLClient *graphql.Client, tenantID string) *FormationAssignmentsAsyncAsserter {
 	f := FormationAssignmentsAsyncAsserter{
 		FormationAssignmentsAsserter: FormationAssignmentsAsserter{
 			expectations:             expectations,
 			expectedAssignmentsCount: expectedAssignmentsCount,
 			certSecuredGraphQLClient: certSecuredGraphQLClient,
 			tenantID:                 tenantID,
 		},
-		delay: delay,
+		timeout: time.Second*8,
+		tick: time.Millisecond*50,
 	}
 	return &f
 }
 
 func (a *FormationAssignmentsAsyncAsserter) AssertExpectations(t *testing.T, ctx context.Context) {
 	formationID := ctx.Value(context_keys.FormationIDKey).(string)
-	a.assertFormationAssignmentsAsynchronously(t, ctx, a.certSecuredGraphQLClient, a.tenantID, formationID, a.expectedAssignmentsCount, a.expectations)
+	a.assertFormationAssignmentsAsynchronouslyWithEventually(t, ctx, a.certSecuredGraphQLClient, a.tenantID, formationID, a.expectedAssignmentsCount, a.expectations)
 }
 
-func (a *FormationAssignmentsAsyncAsserter) assertFormationAssignmentsAsynchronously(t *testing.T, ctx context.Context, certSecuredGraphQLClient *graphql.Client, tenantID, formationID string, expectedAssignmentsCount int, expectedAssignments map[string]map[string]fixtures.AssignmentState) {
-	t.Logf("Sleeping for %d seconds while the async formation assignment status is proccessed...", a.delay)
-	time.Sleep(time.Second * time.Duration(a.delay))
-	listFormationAssignmentsRequest := fixtures.FixListFormationAssignmentRequest(formationID, 200)
-	assignmentsPage := fixtures.ListFormationAssignments(t, ctx, certSecuredGraphQLClient, tenantID, listFormationAssignmentsRequest)
-	require.Equal(t, expectedAssignmentsCount, assignmentsPage.TotalCount)
-
-	assignments := assignmentsPage.Data
-	for _, assignment := range assignments {
-		targetAssignmentsExpectations, ok := expectedAssignments[assignment.Target]
-		require.Truef(t, ok, "Could not find expectations for assignment with ID: %q and target %q", assignment.ID, assignment.Target)
+func (a *FormationAssignmentsAsyncAsserter) assertFormationAssignmentsAsynchronouslyWithEventually(t *testing.T, ctx context.Context, certSecuredGraphQLClient *graphql.Client, tenantID, formationID string, expectedAssignmentsCount int, expectedAssignments map[string]map[string]fixtures.AssignmentState) {
```

Comment:
Could we somehow "mark" or remove the duplicated functions/methods in the new e2e setup? Because this could get messy pretty quickly?

Reply:
During the migrateion process the methods that are no longer used will be removed naturally. The things shouldnt become messy as the asserters and providers are extracted to separate directory in pkg and with time the fixture methods that are in the director tests directory will be migrated there naturally. Even if we mark the duplicated methods now with some comment that will not force us in any way to process them correctly. On the other hand we can not use the methods from the pkg/notifications as they are refactored to have a structure for the new format as receiver.

Reply:
👍 Moreover this is a method and the old assert was a function. That also will help not mixing them.